File: src/main/java/com/example/autotest_backend/service/QuestionService.java
package com.example.autotest_backend.service;

import com.example.autotest_backend.model.Question;
import com.example.autotest_backend.model.QuestionStatus;
import com.example.autotest_backend.model.User;

import java.util.List;

public interface QuestionService {

    Question createQuestion(Question question);

    List<Question> getAllQuestions();

    List<Question> getQuestionsByStatus(QuestionStatus status);

    Question approveQuestion(Long questionId);

    Question rejectQuestion(Long questionId);

    // subjectId is optional — null means "any subject the user belongs to"
    Question getNextQuestion(Long userId, Long subjectId);

    boolean answerQuestion(Long questionId, Long choiceId, User user);
}

File: frontend/src/services/question.service.ts
import api from "./api";
import { CreateQuestionRequest, QuestionResponse } from "../types/question";

export const createQuestion = async (
  data: CreateQuestionRequest
): Promise<QuestionResponse> => {
  const response = await api.post<QuestionResponse>(
    "/api/questions",
    data
  );

  return response.data;
};

export const getQuestionsByStatus = async (status: string) => {
  const response = await api.get(`/api/questions?status=${status}`);
  return response.data;
};

export const getNextQuestion = async (subjectId?: number) => {
  const query = subjectId ? `?subjectId=${subjectId}` : "";
  const response = await api.get(`/api/questions/next${query}`);
  return response.data;
};

export const answerQuestion = async (
  questionId: number,
  choiceId: number
) => {
  const response = await api.post(
    `/api/questions/${questionId}/answer`,
    { choiceId }
  );
  return response.data;
};

export const getCompletedQuestions = async () => {
  const response = await api.get("/api/user-questions/completed");
  return response.data;
};

// Teacher

export const approveQuestion = async (id: number) => {
  const response = await api.patch(
    `/api/questions/${id}/approve`
  );
  return response.data;
};

export const rejectQuestion = async (id: number) => {
  const response = await api.patch(
    `/api/questions/${id}/reject`
  );
  return response.data;
};
